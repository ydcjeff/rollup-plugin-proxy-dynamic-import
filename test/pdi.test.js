import * as fs from 'node:fs';
import * as path from 'node:path';
import { describe, expect, test } from 'vitest';

const ROLLUP_DIST = 'dist/rollup';
const VITE_DIST = 'dist/vite';

/**
 * @param {string} dir
 * @param {string} file
 */
function read(dir, file) {
	return fs.readFileSync(path.resolve(dir, file), 'utf-8');
}

/**
 * @param {string} file
 */
function rollup_read(file) {
	return read(ROLLUP_DIST, file);
}

/**
 * @param {string} file
 */
function vite_read(file) {
	return read(VITE_DIST, file);
}

/**
 * @param {string} name
 * @param {boolean} is_rollup
 * @param {string} file
 */
function run_test(name, is_rollup, file) {
	test(name, () => {
		file = path.join(name, file);
		const content = is_rollup ? rollup_read(file) : vite_read(file);
		expect(content).toMatchSnapshot();
	});
}

/**
 * @param {boolean} is_rollup
 */
function run_all_tests(is_rollup) {
	const file_map = {
		js: ['_js_entry.js', '_js_non_entry.js'],
		js_ext: ['_js_ext_entry.js', '_js_ext_non_entry.js'],
		ts: ['_ts_entry.js', '_ts_non_entry.js'],
	};

	for (const [dir, files] of Object.entries(file_map)) {
		describe(dir, () => {
			for (const file of files) {
				run_test(dir, is_rollup, file);
			}
		});
	}
}

describe('rollup', () => {
	run_all_tests(true);
});

describe('vite', () => {
	run_all_tests(false);
});
