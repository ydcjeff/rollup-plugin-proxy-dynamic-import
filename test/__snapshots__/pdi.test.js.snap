// Vitest Snapshot v1

exports[`rollup > app > app.js 1`] = `
"const js_non_entry = {
	meta: {
		api: () => import('./c/_js.js'),
	},
};

const js_entry = {
	meta: {
		api: () => import('./js/_js_entry.js'),
	},
};

const js_ext_non_entry = {
	meta: {
		api: () => import('./c/_js_ext.js'),
	},
};

const js_ext_entry = {
	meta: {
		api: () => import('./js_ext/_js_ext_entry.js'),
	},
};

const ts_non_entry = {
	meta: {
		api: () => import('./c/_ts.js'),
	},
};

const ts_entry = {
	meta: {
		api: () => import('./ts/_ts_entry.js'),
	},
};

/// <reference types=\\"vite/client\\" />

const globs = import.meta.glob('./import_meta_glob/*.js', {
	eager: true,
	import: 'glob',
});

// eslint-disable-next-line no-console
console.log({
	globs,
	js_entry,
	js_ext_entry,
	js_ext_non_entry,
	js_non_entry,
	ts_entry,
	ts_non_entry,
});
"
`;

exports[`rollup > js > c/_js.js 1`] = `
"function load() {
	return 'js_non_entry_load';
}

const actions = {
	default() {
		return 'js_non_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`rollup > js > js/_js_entry.js 1`] = `
"function load() {
	return 'js_entry_load';
}

const actions = {
	default() {
		return 'js_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`rollup > js_ext > c/_js_ext.js 1`] = `
"function load() {
	return 'js_ext_non_entry_load';
}

const actions = {
	default() {
		return 'js_ext_non_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`rollup > js_ext > js_ext/_js_ext_entry.js 1`] = `
"function load() {
	return 'js_ext_entry_load';
}

const actions = {
	default() {
		return 'js_ext_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`rollup > ts > c/_ts.js 1`] = `
"function load() {
	return 'ts_non_entry_load';
}

const actions = {
	default() {
		return 'ts_non_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`rollup > ts > ts/_ts_entry.js 1`] = `
"function load() {
	return 'ts_entry_load';
}

const actions = {
	default() {
		return 'ts_entry_actions';
	},
};

export { actions, load };
"
`;

exports[`vite > app > app.js 1`] = `
"const page$1 = {
  meta: {
    api: () => import(\\"./c/_page_1.js\\")
  }
};
const page = {
  meta: {
    api: () => import(\\"./c/_page_2.js\\")
  }
};
const pages$1 = /* @__PURE__ */ Object.assign({ \\"./pages/+page_1.js\\": page$1, \\"./pages/+page_2.js\\": page });
const glob$1 = {
  pages: /* @__PURE__ */ Object.values(pages$1),
  meta: {
    api: () => import(\\"./c/_glob.js\\")
  }
};
const pages = /* @__PURE__ */ Object.assign({ \\"./pages/+page_1.js\\": page$1, \\"./pages/+page_2.js\\": page });
const glob = {
  pages: /* @__PURE__ */ Object.values(pages),
  meta: {
    api: () => import(\\"./import_meta_glob/_glob_entry.js\\")
  }
};
const js_non_entry = {
  meta: {
    api: () => import(\\"./c/_js.js\\")
  }
};
const js_entry = {
  meta: {
    api: () => import(\\"./js/_js_entry.js\\")
  }
};
const js_ext_non_entry = {
  meta: {
    api: () => import(\\"./c/_js_ext.js\\")
  }
};
const js_ext_entry = {
  meta: {
    api: () => import(\\"./js_ext/_js_ext_entry.js\\")
  }
};
const ts_non_entry = {
  meta: {
    api: () => import(\\"./c/_ts.js\\")
  }
};
const ts_entry = {
  meta: {
    api: () => import(\\"./ts/_ts_entry.js\\")
  }
};
const globs = /* @__PURE__ */ Object.assign({
  \\"./import_meta_glob/+glob.js\\": glob$1,
  \\"./import_meta_glob/+glob_entry.js\\": glob
});
console.log({
  globs,
  js_entry,
  js_ext_entry,
  js_ext_non_entry,
  js_non_entry,
  ts_entry,
  ts_non_entry
});
"
`;

exports[`vite > import_meta_glob > c/_glob.js 1`] = `
"function load() {
  return \\"glob_non_entry_load\\";
}
const actions = {
  default() {
    return \\"glob_non_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > import_meta_glob > import_meta_glob/_glob_entry.js 1`] = `
"function load() {
  return \\"glob_entry_load\\";
}
const actions = {
  default() {
    return \\"glob_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > js > c/_js.js 1`] = `
"function load() {
  return \\"js_non_entry_load\\";
}
const actions = {
  default() {
    return \\"js_non_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > js > js/_js_entry.js 1`] = `
"function load() {
  return \\"js_entry_load\\";
}
const actions = {
  default() {
    return \\"js_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > js_ext > c/_js_ext.js 1`] = `
"function load() {
  return \\"js_ext_non_entry_load\\";
}
const actions = {
  default() {
    return \\"js_ext_non_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > js_ext > js_ext/_js_ext_entry.js 1`] = `
"function load() {
  return \\"js_ext_entry_load\\";
}
const actions = {
  default() {
    return \\"js_ext_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > ts > c/_ts.js 1`] = `
"function load() {
  return \\"ts_non_entry_load\\";
}
const actions = {
  default() {
    return \\"ts_non_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;

exports[`vite > ts > ts/_ts_entry.js 1`] = `
"function load() {
  return \\"ts_entry_load\\";
}
const actions = {
  default() {
    return \\"ts_entry_actions\\";
  }
};
export {
  actions,
  load
};
"
`;
